firmware/application/ui_external_items_menu_loader.cpp

```diff
@@ -0,0 +1,137 @@
+#include "ui_external_items_menu_loader.hpp"
+
+#include "sd_card.hpp"
+
+namespace ui {
+
+std::vector<Bitmap*> ExternalItemsMenuLoader::bitmaps;
+
+std::vector<GridItem> ExternalItemsMenuLoader::load_external_items(app_location_t app_location, NavigationView& nav) {
+    for (const auto& bitmap : bitmaps) {
+        delete bitmap->data;
+        delete bitmap;
+    }
+    bitmaps.clear();
+
+    std::vector<GridItem> exteral_apps;
+
+    if (sd_card::status() != sd_card::Status::Mounted)
+        return exteral_apps;
+
+    for (const auto& entry : std::filesystem::directory_iterator(u"APPS", u"*.ppma")) {
+        FIL ppma_file;
```

Comment:
Why not use File class?

Reply:
I never really used that thing. I should try it